File: simple_linear_regression.R
# Simple Linear Regression

# Data Preprocessing

# Importing datset
dataset = read.csv('Salary_Data.csv')

#splitting the data into training set and testing set
#install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Fitting Simple Linear Regression to Training Set
regressor = lm(formula = Salary ~ YearsExperience, data = training_set)

# Predicting The Test Set Results
y_pred = predict(regressor, newdata = test_set)

# Visualizing The Trainig Set Results
install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary), 
             color = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary Vs Experience(Trainig Set)') +
  xlab('Years of Experience') +
  ylab('Salary')

# Visualizing The Test Set Results
install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary), 
             color = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary Vs Experience(Test Set)') +
  xlab('Years of Experience') +
  ylab('Salary')

File: simple_linear_regression.py
# -*- coding: utf-8 -*-
"""
Created on Thu Nov 14 12:18:22 2019

@author: pratiksha.garode
"""

# Simple Linear Regression

# Date Preprocessing

# Imprting libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing Datasets
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Splitting the dataset into training set and testing set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Fitting Simple Linear Regression to Training Set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Predicting The Test Set Results
y_pred = regressor.predict(X_test)

# Visualizing The Training Set Results
plt.scatter(X_train, y_train, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary Vs Experience (Training Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

# Visualizing The Test Set Results
plt.scatter(X_test, y_test, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary Vs Experience (Test Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()











